webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js

```diff
@@ -44,11 +44,14 @@ export const loadSwitcherResourcesByResource = (resource, { page = 1, searchQuer
     dispatch({ type: BREADCRUMB_BAR_RESOURCES_FAILURE, payload: { error, resourceUrl } });
 
   const formatResults = ({ data }) => {
-    const switcherItems = flatten(Object.values(data.results)).map(result => ({
-      name: get(result, nameField),
-      id: result.id,
-      url: switcherItemUrl.replace(':id', result.id),
-    }));
+    const switcherItems = flatten(Object.values(data.results)).map((result) => {
+      const itemName = get(result, nameField);
+      return ({
+        name: get(result, nameField),
+        id: result.id,
+        url: switcherItemUrl.replace(':id', result.id).replace(':name', itemName),
```

Comment:
instead calling `replace` twice, we can do something like this:
```javascript
const mapTemplateVariable = {
   ':id': result.id,
   ':name': itemName,
};
return ({
  url: switcherItemUrl.replace(/:id|:name/gi, (matched) =>  mapTemplateVariable[matched]);
})
```
With that we can extend item url template with ease later on.

Reply:
IMHO using `replace` looks better for readability and easier to understand.

Reply:
@amir @sharvit maybe leave it more readable now, and when we grow with the number of template variables we"ll change to a mapper?